src/components/PinInput/src/PinInput.vue

```diff
@@ -0,0 +1,308 @@
+		 * Lenth of pin
+		 */
+		pinLength: {
+			type: Number,
+			default: DEFAULT_INPUT_SIZE,
+			validator: (value) => value > 0,
+		},
+
+		/**
+		 * Input variant
+		 */
+		variant: {
+			type: String,
+			default: 'fill',
+			validator: (variant) => ['fill', 'outline'].includes(variant),
+		},
+
+		/**
+		 * Toggles input invalid state
+		 */
+		invalid: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * Disable the inputs
+		 */
+		disabled: {
+			type: Boolean,
+			default: false,
+		},
+	},
+
+	data() {
+		return {
+			code: new Array(this.pinLength).fill(''),
+			isShaking: false,
+		};
+	},
+
+	computed: {
+		currentCode() {
+			return this.code.join('');
+		},
+
+		containerWidth() {
+			const SINGLE_CONTAINER_WIDTH = 58;
+			return `${this.pinLength * SINGLE_CONTAINER_WIDTH}px`;
+		},
+	},
+
+	watch: {
+		// Refocus on first input when re-enabled
+		disabled(isDisabled) {
+			if (!isDisabled) {
+				this.focusOnCodeCell(0);
+			}
+		},
+	},
+
+	mounted() {
+		this.focusOnCodeCell(0);
+	},
+
+	methods: {
+		findFirstIncompleteCodeCellIndex() {
+			return this.code.findIndex((value) => value === '');
+		},
+
+		getCodeCellRef(index) {
+			return `code-cell-${index}`;
+		},
+
+		// Focus on verification code cell input at given index.
+		focusOnCodeCell(index) {
+			const cellReference = this.$refs[this.getCodeCellRef(index)];
+			if (cellReference?.[0]?.focus) {
+				cellReference[0].focus();
+			}
+		},
+
+		onInputCode(event, index) {
+			if (!event.data) {
+				return;
+			}
+
+			// Only allow integers in input
+			const BASE_TEN = 10;
+			const inputValue = Number.isInteger(Number.parseInt(event.data, BASE_TEN)) ? event.data : '';
+
+			// One-time-code autofill is treated as an input, not a paste
+			if (this.attemptSplitCodeIntoInputs(inputValue, index)) {
+				return;
+			}
+
+			this.$set(this.code, index, inputValue);
+
+			const firstIncompleteCellIndex = this.findFirstIncompleteCodeCellIndex();
+
+			// eslint-disable-next-line no-magic-numbers
+			if (firstIncompleteCellIndex === -1) {
+				this.handleComplete(this.currentCode);
+				return;
+			}
+
+			// This allows us to auto jump to the next code cell input as the user types.
+			this.focusOnCodeCell(firstIncompleteCellIndex);
+		},
+
+		attemptSplitCodeIntoInputs(value, inputIndex) {
+			const BASE_TEN = 10;
+			if (inputIndex === 0
+				&& value?.length === this.pinLength
+				&& Number.isInteger(Number.parseInt(value, BASE_TEN))
+			) {
+				this.$set(this, 'code', value.split(''));
+
+				// Having a timeout here gives the user a chance to see their code before success/failure
+				const TIMEOUT_LENGTH_MS = 500;
+				setTimeout(() => { this.handleComplete(this.currentCode); }, TIMEOUT_LENGTH_MS);
+				return true;
+			}
+			return false;
+		},
+
+		handleComplete(code) {
+			this.$emit('complete', code);
+		},
+
+		onPasteCode(event, index) {
+			event.preventDefault();
+			const pastedValue = (event.clipboardData || window.clipboardData).getData('text');
```

Comment:
just asking to double-check, but is `(event.clipboardData || window.clipboardData)` a reliable way to get the clipboard data on all of our supported browsers, i.e. Chrome, Firefox, Safari, Edge & all of their mobile versions?

Reply:
seems to be reliable yeah